regression/dataset.py

```diff
@@ -0,0 +1,126 @@
+# Copyright 2023 MosaicML Streaming authors
```

Comment:
Can we rename this to `synthetic_dataset.py` to avoid confusion with other dataset.py file ?